clients/typescript/src/types.ts

```diff
@@ -1,19 +1,89 @@
+/**
+ * @typedef {Record<string, unknown>} StanVariableInputs
+ * A type holding named inputs to a Stan model,
+ * e.g. the data or initial values.
+ */
 export type StanVariableInputs = Record<string, unknown>;
 
+/**
+ * @typedef {Function} PrintCallback
+ * A callback for printing output from the Stan model.
+ * @param {string} s The string to print.
+ * @returns {void}
+ */
 export type PrintCallback = (s: string) => void;
 
+/**
+ * The metric used for the HMC sampler.
+ * @enum {number}
```

Comment:
As an `enum` and not technically a `typedef`, this doesn't show up anywhere in the Table-of-Contents section that's generated in `main`. IDK if you care or how much of a hassle it would be to change that, just noting it.

Reply:
Good spot, but also good intuition that this will probably be too annoying to be worth fixing